lib/documents.js

```diff
@@ -186,6 +199,44 @@ export async function register({
     }
   }
 }
+/**
+* Raises the minimum assurance for resolution on both the registration and
+* token batches.
+*
+* @param {object} options - Options to use.
+* @param {string} options.externalId - An application-specific string that
+*   unambiguously identifies a particular entity to the application.
+* @param {object} options.document - The document to register.
+* @param {string} [options.creator] - Optional identifier for the creator
+*   of the registration.
+* @param {number} options.ttl - The number of milliseconds until the
+*   document should expire.
+* @param {number} options.minAssuranceForResolution - Minimum level of
+*   assurance required for token resolution.
+*
+* @returns {Promise<Buffer>} The internalId of the registration record.
+*/
+export async function raiseAssurance({
+  externalId, document, creator, ttl, minAssuranceForResolution
+} = {}) {
+  const tokenizer = await tokenizers.getCurrent();
+
+  const {externalIdHash, documentHash, creatorHash} =
+      await _tokenizeRegistration({tokenizer, externalId, document, creator});
+
+  const {registration} =
+    await _getRegistrationRecord({externalIdHash, documentHash});
+
+  await _refresh({
+    externalIdHash, documentHash, ttl, creatorHash, minAssuranceForResolution,
+  });
+
+  await updateBatch({
+    internalId: registration.internalId, minAssuranceForResolution
+  });
```

Comment:
We need a large comment block above all of these steps that details how it won't create atomicity issues -- and where the potential pitfalls are, if any.

Reply:
I'm also concerned about what happens when this function is called and the registration has expired. The comments should explain how that isn't an issue with details for why.

Reply:
This has been refactored, and a comment has been added to indicate if it gets to that point it cannot be expired since it will have just been refreshed in the `if`.